tests/lib/prepare.sh

```diff
@@ -524,18 +524,161 @@ EOF
     rm -rf "$UNPACK_DIR"
 }
 
-
-uc20_build_initramfs_kernel_snap() {
+install_core_initrd_deps() {
     # carries ubuntu-core-initframfs
     add-apt-repository ppa:snappy-dev/image -y
     # TODO: install the linux-firmware as the current version of
     # ubuntu-core-initramfs does not depend on it, but nonetheless requires it
     # to build the initrd
     apt install ubuntu-core-initramfs linux-firmware -y
+}
+
+uc22_build_initramfs_kernel_snap() {
+    local ORIG_SNAP="$1"
+    local TARGET="$2"
+
+    install_core_initrd_deps
+
+    # TODO proper option support here would be nice but bash is hard and this is
```

Comment:
could we combine more of the shared bits of uc22_build_initramfs_kernel_snap and uc20_build_initramfs_kernel_snap here and then have uc20_build_initramfs_kernel_snap call another helper to modify the-tool since uc22_build_initramfs_kernel_snap doesn't need it?

Reply:
done